src/controllers/signMessage/signMessage.test.ts

```diff
@@ -37,106 +37,74 @@ const account: Account = {
   }
 }
 
-let accountStates: AccountStates
-
-const getAccountsInfo = async (accounts: Account[]): Promise<AccountStates> => {
-  const result = await Promise.all(
-    networks.map((network) => getAccountState(providers[network.id], network, accounts))
-  )
-  const states = accounts.map((acc: Account, accIndex: number) => {
-    return [
-      acc.addr,
-      Object.fromEntries(
-        networks.map((network: Network, netIndex: number) => {
-          return [network.id, result[netIndex][accIndex]]
-        })
-      )
-    ]
-  })
-  return Object.fromEntries(states)
+const messageToSign: Message = {
+  fromActionId: 1,
+  content: { kind: 'message', message: '0x74657374' },
+  accountAddr: account.addr,
+  signature: null,
+  networkId: 'ethereum'
 }
 
 describe('SignMessageController', () => {
   let signMessageController: SignMessageController
   let keystore: KeystoreController
 
-  beforeAll(async () => {
-    accountStates = await getAccountsInfo([account])
-  })
-
   beforeEach(async () => {
-    const keystoreSigners = { internal: InternalSigner }
-    keystore = new KeystoreController(produceMemoryStore(), keystoreSigners)
+    const storage = produceMemoryStore()
+    await storage.set('accounts', JSON.stringify([account]))
+    await storage.set('selectedAccount', JSON.stringify(account.addr))
+
+    keystore = new KeystoreController(storage, { internal: InternalSigner })
     let providersCtrl: ProvidersController
     const networksCtrl = new NetworksController(
-      produceMemoryStore(),
+      storage,
       fetch,
       (net) => {
         providersCtrl.setProvider(net)
       },
-      (id) => {
-        providersCtrl.removeProvider(id)
-      }
+      (id) => providersCtrl.removeProvider(id)
     )
     providersCtrl = new ProvidersController(networksCtrl)
     providersCtrl.providers = providers
 
-    const mockedAccountsCtrl = {
-      accountStates,
-      accounts: [account]
-    } as AccountsController
+    const accountsCtrl = new AccountsController(
+      storage,
+      providersCtrl,
+      networksCtrl,
+      () => {},
+      () => {}
+    )
+    // Internally, it waits Networks and Providers to be loaded
+    await accountsCtrl.initialLoadPromise
+
+    await accountsCtrl.updateAccountState(account.addr, 'latest')
```

Comment:
same here: do not "await"

Reply:
Yes, no longer needed after the changes I mentioned in https://github.com/AmbireTech/ambire-common/pull/880#discussion_r1679079120